Assets/_System/Scripts/TurnBasedScripts/BaseUnitClass.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public enum UnitType { Swordsman}
+
+public abstract class BaseUnitClass : GridObject
+{
+
+    [Header("Unit Type")]
+    public UnitType type;
```

Comment:
If we make this public then other things can modify it. We probably don't want that.

Instead do : 

```
[Header("Unit Type")]
[SerializeField]
private UnitType type;

public UnitType Type => type;

```